src/Utility/AnaSol.C

```diff
@@ -17,13 +17,54 @@
 #include "Utilities.h"
 #include "IFEM.h"
 #include "tinyxml.h"
+
+#include <autodiff/reverse/var.hpp>
+
 #ifdef HAS_HDF5
-#include "ProcessAdm.h"
-#include "HDF5Writer.h"
 #include "FieldFunctions.h"
 #endif
 
 
+namespace {
+
+//! \brief Function returning the gradient of a scalar or vector function.
+template<class FromFunc, class ToFunc, class Ret>
+class GradientFunc : public ToFunc
+{
+public:
+  //! \brief The constructor initializes the function to use.
+  GradientFunc(const FromFunc& f) : func(f) {}
+
+protected:
+  //! \brief Evaluates the gradient in a point.
+  Ret evaluate(const Vec3& X) const
+  {
+    return func.gradient(X);
+  }
```

Comment:
This single-statement method can also be written on one line.

Reply:
Ok